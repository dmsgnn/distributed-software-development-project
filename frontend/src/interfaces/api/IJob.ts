import type { IRepo } from '@/interfaces/api/IRepo';

interface IGitLeaksJob {
	results: Array<{
		author: string;
		commit: string;
		date: string;
		description: string;
		email: string;
		endColumn: number;
		endLine: number;
		entropy: number;
		file: string;
		fingerprint: string;
		match: string;
		message: string;
		ruleID: string;
		secret: string;
		startColumn: number;
		startLine: string;
		symlinkFile: string;
	}>;
}

interface IGoKartJob {
	results: Array<{
		sinkColumn: number;
		sinkFile: string;
		sinkLine: number;
		sinkName: string;
		sourceColumn: Array<number>;
		sourceFile: Array<string>;
		sourceLine: Array<number>;
		sourceName: Array<string>;
		vulnCwe: string;
		vulnId: string;
		vulnName: string;
		vulnType: string;
	}>;
}

interface IProgPilotJob extends IGoKartJob {}

interface IBanditJob {
	errors: Array<{
		filename: string;
		reason: string;
	}>;
	metrics: Array<{
		confidenceHigh: number;
		confidenceLow: number;
		confidenceMedium: number;
		confidenceUndefined: number;
		severityHigh: number;
		severityLow: number;
		severityMedium: number;
		severityUndefined: number;
		loc: number;
		nosec: number;
		skippedTests: number;
	}>;
	results: Array<{
		sinkColumn: number;
		sinkFile: string;
		sinkLine: number;
		sinkName: string;
		sourceColumn: Array<number>;
		sourceFile: Array<string>;
		sourceLine: Array<number>;
		sourceName: Array<string>;
		vulnCwe: string;
		vulnId: string;
		vulnName: string;
		vulnType: string;
	}>;
}

export interface IJob {
	id: number;
	startTime: string;
	endTime?: string;
	compliant: boolean;
	log?: IGitLeaksJob | IGoKartJob | IProgPilotJob | IBanditJob;
	repo: IRepo;
}
