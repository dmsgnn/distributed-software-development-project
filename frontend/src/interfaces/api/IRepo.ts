export interface IRepo {
	id: number;
	fullName: string;
	url: string;
	htmlUrl: string;
	hooksUrl: string;
	hookUrl: string;
	branchesUrl: string;
	cloneUrl: string;
	defaultBranch: string;
}
