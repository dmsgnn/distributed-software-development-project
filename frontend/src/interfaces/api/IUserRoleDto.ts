export interface IUserRoleDto {
	id: number;
	roleName: string;
}
