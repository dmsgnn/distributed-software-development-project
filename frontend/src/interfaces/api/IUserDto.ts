import { IUserRoleDto } from '@/interfaces/api/IUserRoleDto';

export interface IUserDto {
	id: number;
	username: string;
	email: string;
	firstName: string;
	lastName: string;
	userRole: IUserRoleDto;
}
