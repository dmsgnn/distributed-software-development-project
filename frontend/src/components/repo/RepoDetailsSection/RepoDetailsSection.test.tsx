import { IRepo } from '@/interfaces/api/IRepo';
import { axe } from 'jest-axe';

import { act, render, screen, waitForElementToBeRemoved } from 'test-utils';
import { RepoDetailsSection } from './';

jest.mock('next/router', () => ({
	useRouter: () => ({
		query: { repoId: '1' },
	}),
}));

describe('RepoDetailsSection', () => {
	describe('when repoId is falsy', () => {
		it('should matches snapshot', () => {
			const { asFragment } = render(<RepoDetailsSection repoId="" />);

			expect(asFragment()).toMatchSnapshot();
		});

		it('should be accessible', async () => {
			const { container } = render(<RepoDetailsSection repoId="" />);

			let results: Awaited<ReturnType<typeof axe>>;
			await act(async () => {
				results = await axe(container);
			});
			expect(results).toHaveNoViolations();
		});
	});

	describe('when repoId exists', () => {
		beforeEach(() => {
			window.fetch = jest.fn().mockImplementation(() => {
				return Promise.resolve({
					ok: true,
					status: 200,
					headers: new Headers({ 'Content-Type': 'application/json', 'Content-Length': '1' }),
					json: () =>
						Promise.resolve({
							id: 1,
							name: 'test',
							description: 'test',
							owner: {
								login: 'test',
								avatar_url: 'test',
								html_url: 'test',
							},
							html_url: 'test',
							language: 'test',
							stargazers_count: 1,
							created_at: 'test',
							privacy: 1,
							security: 2,
						} as Partial<IRepo>),
				});
			});
		});

		it('should matches snapshot', async () => {
			const { asFragment } = render(<RepoDetailsSection repoId="1" />);

			await waitForElementToBeRemoved(() => screen.getByText('Loading...'));

			expect(asFragment()).toMatchSnapshot();
		});

		it('should be accessible', async () => {
			const { container } = render(<RepoDetailsSection repoId="1" />);

			let results: Awaited<ReturnType<typeof axe>>;
			await act(async () => {
				results = await axe(container);
			});
			expect(results).toHaveNoViolations();
		});

		afterEach(() => {
			jest.restoreAllMocks();
		});
	});
});
