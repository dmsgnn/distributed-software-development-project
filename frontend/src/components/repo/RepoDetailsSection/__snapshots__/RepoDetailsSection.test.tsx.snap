// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoDetailsSection when repoId exists should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <h2
      class="chakra-heading css-0"
    />
    <div
      class="css-pn2qr0"
    >
      <span
        class="css-1qm7lvz"
      >
        Type: 
      </span>
      <span
        class="css-1qm7lvz"
      >
        Domain: 
      </span>
      <span
        class="css-1qm7lvz"
      >
        User data: ‚ùå
      </span>
      <span
        class="css-1qm7lvz"
      >
        Privacy: 1 / 6
      </span>
      <span
        class="css-1qm7lvz"
      >
        Security: 2 / 6
      </span>
      <span
        class="css-1qm7lvz"
      >
        <span
          class="css-zvlevn"
        >
          <a
            class="chakra-link css-0"
            rel="noopener"
            target="_blank"
          >
            GitHub Repo
          </a>
        </span>
        <svg
          class="chakra-icon chakra-icon css-1r2vfnw"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-width="2"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <path
              d="M15 3h6v6"
            />
            <path
              d="M10 14L21 3"
            />
          </g>
        </svg>
      </span>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        Delete repo
      </button>
    </div>
    <p
      class="chakra-text css-9rlny8"
    >
      Repo description
    </p>
    <p
      class="chakra-text css-cssgv9"
    >
      test
    </p>
  </div>
</DocumentFragment>
`;

exports[`RepoDetailsSection when repoId is falsy should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-gmuwbf"
  >
    <div
      class="chakra-spinner css-1y7joxr"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
</DocumentFragment>
`;
